File: WEEK21/그래프_가장 먼 노드.py
from collections import deque

def solution(n, edge):
    vertices = sorted(edge)
    graph = [[] for _ in range(n + 1)]
    route = [0] * (n + 1)
    
    for vertex in vertices:
        graph[vertex[0]].append(vertex[1])
        graph[vertex[1]].append(vertex[0])
    
    queue = deque()
    queue.append(1)
    route[1] = 1
    
    while queue:
        node = queue.popleft()
        
        for i in graph[node]:
            if route[i] == 0:
                queue.append(i)
                route[i] = route[node] + 1
    
    answer = 0
    max_distance = max(route)
    
    for j in route:
        if j == max_distance:
            answer += 1
    
    return answer

File: WEEK21/그래프_가장 먼 노드.js
function solution(n, edge) {
  let visited = new Array(n + 1).fill(-1);
  let queue = new Array();
  let adjacent = new Array(n + 1).fill(null).map(() => Array());

  visited[0] = 0;
  visited[1] = 0;
  queue.push(1);

  for (let i of edge) {
    adjacent[i[0]].push(i[1]);
    adjacent[i[1]].push(i[0]);
  }

  while (queue.length) {
    let node = queue.shift();

    if (adjacent[node]) {
      adjacent[node].forEach((element) => {
        if (visited[element] === -1) {
          queue.push(element);
          visited[element] = visited[node] + 1;
        }
      });
    }
  }

  let maxDistance = Math.max(...visited);
  let answer = visited.filter((element) => element === maxDistance).length;

  return answer;
}
